aodncore/pipeline/steps/harvest.py

```diff
@@ -181,25 +193,36 @@ def execute_talend(self, executor, matched_file_list, talend_base_dir):
                 p.execute()
             except SystemCommandFailedError:
                 self._logger.error(p.stdout_text)
-                # Attempt to delete the files that have already been harvested by one or more previous harvesters
-                self.cleanup_on_error()
                 raise
-            else:
-                self._logger.info(p.stdout_text)
             finally:
                 self._logger.info('--- END TALEND OUTPUT ---')
 
-    def run_deletions(self, harvester_map, tmp_base_dir):
+    def run_deletions(self, harvester_map, tmp_base_dir, deletion_attribute):
         for harvester, matched_files in harvester_map.items():
             with TemporaryDirectory(prefix='talend_base', dir=tmp_base_dir) as talend_base_dir:
-                matched_file_list = [mf.dest_path for mf in matched_files]
-
-                self.execute_talend(self._config.trigger_config[harvester]['exec'], matched_file_list, talend_base_dir)
+                for pf in matched_files:
+                    self.remove_symlink(talend_base_dir, pf.dest_path)
+                try:
+                    self.execute_talend(self._config.trigger_config[harvester]['exec'], matched_files, talend_base_dir)
+                except SystemCommandFailedError:
+                    raise
 
-            for mf in matched_files:
-                mf.is_harvested = True
+            files_to_delete = matched_files.filter_by_bool_attributes_and(deletion_attribute)
+            if files_to_delete:
+                self.upload_runner.run(files_to_delete)
 
-            files_to_delete = matched_files.filter_by_bool_attribute('pending_store_deletion')
+    def run_undo_deletions(self, harvester_map, tmp_base_dir, deletion_attribute):
+        for harvester, matched_files in harvester_map.items():
+            with TemporaryDirectory(prefix='talend_base', dir=tmp_base_dir) as talend_base_dir:
+                for pf in matched_files:
+                    self.remove_symlink(talend_base_dir, pf.dest_path)
```

Comment:
Just replace this with os.remove. It's running out of the temp dir, and the files should definitely exist, so let it throw unhandled, since it would be a very dodgy bug scenario.